### PR TITLE
chitter challenge

### PR DESCRIPTION
**Pull request by jshorns**

### CODE REVIEW

File: app.rb [hunk 1]
```diff
@@ -0,0 +1,73 @@
...
+    @content = params[:content]
+    time = DateTime.now
+    @date = time.strftime("%Y-%m-%d")
+    @time = time.strftime("%H:%M")
+    DBConnection.query("INSERT INTO peeps (content, date, time, user_id) VALUES('#{@content}', '#{@date}', '#{@time}', '#{@user_id}');")
+    redirect('/')
+  end
+
+  run! if app_file == $0
+end
```
Comment (Laura-Keen) [inline app.rb:73]: Great coverage of user stories and edge cases. Clearly thought a lot about how the app will be used  whilst keeping code clean and easy to follow!

File: db/migrations/01_create_user_table.sql [hunk 2]
```diff
@@ -0,0 +1,15 @@
...
+  username VARCHAR(20))
+
+CREATE TABLE peeps (
+  id SERIAL PRIMARY KEY,
+  content VARCHAR(140),
+  date DATE,
+  time TIME,
+  user_id INT,
+  FOREIGN KEY (user_id) REFERENCES users (id)
+);
```
Comment (Laura-Keen) [inline db/migrations/01_create_user_table.sql:15]: Good use of clear instructions for another developer to easily set up the project

File: Gemfile [hunk 3]
```diff
@@ -1,6 +1,18 @@
...
+group :security do
+  gem 'bcrypt'
+end
+
+group :development do
+  gem 'pg'
+  gem 'sinatra'
+  gem 'sinatra-flash'
+end
+
```
Comment (Laura-Keen) [inline Gemfile:12]: Really like the use of grouping gems together

File: spec/spec_helper.rb [hunk 4]
```diff
@@ -1,5 +1,21 @@
...
+require 'capybara'
+require 'rspec'
+require 'capybara/rspec'
 require 'simplecov'
 require 'simplecov-console'
+require 'setup_test_database'
+require 'web_helpers'
+require 'database_helpers'
+
+RSpec.configure do |config|
```
Comment (Laura-Keen) [inline spec/spec_helper.rb:12]: Nice clearing of test data base before each use

File: views/users/new.erb [hunk 5]
```diff
@@ -0,0 +1,12 @@
...
+  <label for "email">Email:</label>
+  <input type="text" id="email" name="email" maxlength="40"/>
+  <label for "password">Password:</label>
+  <input type="password" id="password" name="password" maxlength="20"/>
+  <label for "name">Name:</label>
+  <input type="text" id="name" name="name" maxlength="50"/>
+  <label for "username">Username:</label>
+  <input type="text" id="username" name="username" maxlength="20"/>
+  <input type="submit" value="Submit"/>
+</form>
```
Comment (Laura-Keen) [inline views/users/new.erb:12]: Nice to see everything clearly labelled. User friendly layout, clear to follow

**Review by Laura-Keen — COMMENTED** [review]
Really impressed with this project! Clean, DRY code that is easy for another developer to follow. I like that you have given clear instructions and SQL commands. Good test coverage and you have definitely thought about all of the user requirements and edge cases which comes across in both the code and the tests
